### PR TITLE
[Wikipedia] Multi Language Support

### PR DESCRIPTION
**Pull request by JonathanWbn**
## Description

- adds a language dropdown to select from the [sixth most active Wikipedia editions](https://en.wikipedia.org/wiki/List_of_Wikipedias#Basic_list). The current extension only searches in the English edition.
- remember last-selected language with `useCachedState`
- refactored data fetching using Raycast utils instead of `swr`
- updated dependencies & removed obsolete dependencies
- solves #3958

## Screencast

https://user-images.githubusercontent.com/19997520/208733903-aea723d2-ea9a-42ff-a180-dff2f9452438.mov

### CODE REVIEW

File: extensions/wikipedia/src/search-page.tsx [hunk 1]
```diff
@@ -1,48 +1,54 @@
...
 
   return (
     <List
       throttle
-      isLoading={isValidating}
+      isLoading={isLoading}
       onSearchTextChange={setSearch}
       searchBarPlaceholder="Search pages by name..."
+      searchBarAccessory={
+        <List.Dropdown tooltip="Language" value={language} onChange={setLanguage}>
```
Comment (pernielsentikaer) [inline extensions/wikipedia/src/search-page.tsx:19]: Maybe you can use [storeValue property](https://developers.raycast.com/api-reference/user-interface/list#props-1) instead?

**Comment by pernielsentikaer** [comment]
Hi @vimtor 👋 

Do you mind checking this too

**Comment by laune-** [comment]
Thank you for your contribution. I hope this will be merged fast into the main branch :)

File: extensions/wikipedia/src/search-page.tsx (hunk 1 above)
Comment (JonathanWbn) [reply extensions/wikipedia/src/search-page.tsx]: I've applied that change here: https://github.com/raycast/extensions/pull/3976/commits/59f29803c6129dddfce5a2385efc37817ecfa7bf

Is this the approach you would recommend?

**Comment by vimtor** [comment]
@JonathanWbn thank you for the contribution and for updating the deps. I approve the change, however I wonder if it would make more sense to have the language as a preference instead.

You will know the use case better so I let it up to you to decide :)

**Comment by JonathanWbn** [comment]
> thank you for the contribution and for updating the deps. I approve the change, however I wonder if it would make more sense to have the language as a preference instead.

@vimtor thanks!
I prefer the search bar dropdown for the following reasons:
- better UX: no initial config prompt, comfortable to change after the initial setting
- solves the bi-lingual use case: the original issue reads "in **most cases** i want to search in the german wikipedia, but autosuggestion only works for the english wiki." (#3958)

**Comment by vimtor** [comment]
> > thank you for the contribution and for updating the deps. I approve the change, however I wonder if it would make more sense to have the language as a preference instead.
> 
> @vimtor thanks! I prefer the search bar dropdown for the following reasons:
> 
> * better UX: no initial config prompt, comfortable to change after the initial setting
> * solves the bi-lingual use case: the original issue reads "in **most cases** i want to search in the german wikipedia, but autosuggestion only works for the english wiki." ([[Wikipedia] ... #3958](https://github.com/raycast/extensions/issues/3958))

The 2nd point makes a lot of sense. Let's ship it @pernielsentikaer! 🚀

**Review by pernielsentikaer — APPROVED** [review]

**Comment by NikosEfthias** [comment]
> @JonathanWbn thank you for the contribution and for updating the deps. I approve the change, however I wonder if it would make more sense to have the language as a preference instead.
> 
> You will know the use case better so I let it up to you to decide :)

I also agree with the author. I do use wikipedia often in multiple languages it would be a PIA to go to settings and change every time the setting. I think the authors design is really useful. 
PS. I would like  to add turkish and greek languages shall i create a PR or if it is something too minor for a pr would you add it?
@vimtor @pernielsentikaer

**Comment by vimtor** [comment]
> > @JonathanWbn thank you for the contribution and for updating the deps. I approve the change, however I wonder if it would make more sense to have the language as a preference instead.
> > You will know the use case better so I let it up to you to decide :)
> 
> I also agree with the author. I do use wikipedia often in multiple languages it would be a PIA to go to settings and change every time the setting. I think the authors design is really useful. PS. I would like to add turkish and greek languages shall i create a PR or if it is something too minor for a pr would you add it? @vimtor @pernielsentikaer

Yeah! Feel free to create a PR with those languages

**Comment by NikosEfthias** [comment]
thanks @vimtor I submit my pr directly to raycast repo right ? I couldnt see a contributing md file or something thats why asking here

**Comment by pernielsentikaer** [comment]
You can read a bit [more here](https://developers.raycast.com/basics/contribute-to-an-extension) @NikosEfthias

**Comment by NikosEfthias** [comment]
@pernielsentikaer thank you

**Pull request merged**